### PR TITLE
Add offline indicator

### LINKED ISSUE #472: Nheko doesn't show an error message when there is no internet
When there is no internet, Nheko doesn't show any error message. It would be great if Nheko could display one when the connection drops.

Also, some messages don't send after internet has been restored.
Example:
If I send a message when the connection goes down, the message still doesn't deliver even when the internet comes back.

Thanks

### PR DESCRIPTION
**Pull request by LorenDB**
Closes #472﻿

### CODE REVIEW

File: src/timeline/TimelineViewManager.h [hunk 1]
```diff
@@ -45,6 +45,7 @@ class TimelineViewManager : public QObject
       bool isInitialSync MEMBER isInitialSync_ READ isInitialSync NOTIFY initialSyncChanged)
     Q_PROPERTY(
       bool isWindowFocused MEMBER isWindowFocused_ READ isWindowFocused NOTIFY focusChanged)
+    Q_PROPERTY(bool hasConnection READ hasConnection NOTIFY hasConnectionChanged)
```
Comment (deepbluev7) [inline src/timeline/TimelineViewManager.h:48]: `isConnected` sounds better imo

**Comment by deepbluev7** [comment]
I'm not really a fan of a red bar across the whole client... I think I would prefer having a red bar below the profile section, but that won't notify you in mobile mode, I guess... I guess this is fine, but it seems a bit too obnoxious xD

**Comment by LorenDB** [comment]
I agree that it could be obnoxious, but yes, I specifically wanted to make sure it was visible everywhere ¯\_(ツ)_/¯

File: resources/qml/ChatPage.qml [hunk 2]
```diff
@@ -17,110 +17,132 @@ Rectangle {
...
+        Rectangle {
+            id: offlineIndicator
 
-        Connections {
-            target: Communities
-            function onCurrentTagIdChanged() {
-                adaptiveView.initializePageIndex();
-            }
-        }
+            z: 3
```
Comment (deepbluev7) [inline resources/qml/ChatPage.qml:27]: Why?

File: resources/qml/ChatPage.qml [hunk 3]
```diff
@@ -17,110 +17,132 @@ Rectangle {
...
+            color: Nheko.theme.error
+            visible: !TimelineManager.hasConnection
+            Layout.preferredHeight: fontMetrics.height + Nheko.paddingMedium
```
Comment (deepbluev7) [inline resources/qml/ChatPage.qml:30]: I suggest you set this to the labelHeight + padding

File: resources/qml/ChatPage.qml (hunk 2 above)
Comment (LorenDB) [reply resources/qml/ChatPage.qml:27]: Artifact from earlier implementations

**Comment by LorenDB** [comment]
Ready to roll!

**Comment by deepbluev7** [comment]
ty <3

**Pull request merged**